### PR TITLE
Test merging branches

### PR DESCRIPTION
**Pull request by bryanduerk**
This merges all pull requests into one set, except for elevator.

### CODE REVIEW

File: src/org/usfirst/frc/team467/robot/Drive.java [hunk 1]
```diff
@@ -106,8 +106,8 @@ public void zero() {
 	}
 
 	public void sendData() {
-		RobotData.getInstance().update(getRightDistance(), getLeftDistance());
-	}
+		RobotData.getInstance().updateDrivePosition(getRightDistance(), getLeftDistance());
+	} // changed update to updateDrivePosition
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: This comment will only confuse future readers. There's no need to comment on what changes, just comment on what is there now.

File: src/org/usfirst/frc/team467/robot/Autonomous/MatchConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+	// Simulator variables
+	private String simulatedGameSpecificMessage = "LLL";
+
+	private Alliance simulatedTeamColor = Alliance.Red;
+
+	private String simulatedAutoMode = "Left";
+
+	private static MatchConfiguration instance;
+
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Autonomous/MatchConfiguration.java]: Collapse the blank lines between variables.

File: src/org/usfirst/frc/team467/robot/Autonomous/MatchConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+	private TeamColor teamColor;
+
+	private Side redSwitch;
+
+	private Side blueSwitch;
+
+	private Side scale;
+
+	private StartPosition startPosition;
+
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Autonomous/MatchConfiguration.java:55]: Like above, don't put in all these blank lines between variables.

File: src/org/usfirst/frc/team467/robot/Autonomous/MatchConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+		if(color == DriverStation.Alliance.Blue) {
+			LOGGER.info("Alliance is blue");
+			teamColor = TeamColor.BLUE;
+		} else if (color == DriverStation.Alliance.Red){
+			LOGGER.info("Alliance is red");
+			teamColor = TeamColor.RED;
+		} else {
+			LOGGER.info("Alliance not found");
+			teamColor = TeamColor.UNKNOWN;
+		}
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Autonomous/MatchConfiguration.java]: You should code this using a switch statement just like in setAutoModeAndStartPosition. They are basically doing the same thing but doing it differently. Either approach is good, but try to be consistent.

File: src/org/usfirst/frc/team467/robot/DriverStation.java [hunk 5]
```diff
@@ -86,11 +86,6 @@ public double getTurnSensivity() {
 		return 0.0;
 	}
 
-	public void periodic() {
-		driverRumbler.periodic();
-		navRumbler.periodic();
-	}
-
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/DriverStation.java]: I think it works better with the design we're aligning on if DriverStation has a periodic function -- I'd like to add it back and then remove the rumbler periodics from readInputs.

File: src/org/usfirst/frc/team467/robot/DriverStation.java [hunk 6]
```diff
@@ -170,31 +165,11 @@ public void driverSetRightRumble(double value) {
 		driverJoy.rightRumble(value);
 	}
 
-	public void setDriverRumble(double value) {
-		getDriveJoystick().setRumble(value);
+	public void setDriverRumble(int rumbleDuration) {
+		driverRumbler.rumble(rumbleDuration, 1);
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/DriverStation.java]: This isn't right. The caller should call getDriverRumbler() and then ask the rumbler directly to do the work.

In fact, on master branch setDriverRumble() is no longer called by anyone and should be removed.

File: src/org/usfirst/frc/team467/robot/DriverStation.java [hunk 7]
```diff
@@ -170,31 +165,11 @@ public void driverSetRightRumble(double value) {
...
-	/**
-	 * Set a value in 'Basic' tab of the driver station
-	 * 
-	 * @param slot position of the value to set (0-9)
-	 * @param value any integer value
-	 */
-	public void set(int slot, int value) {
-		set(slot, String.valueOf(value));
+	public void setNavRumble(int rumbleDuration) {
+		this.navRumbler.rumble(rumbleDuration, 1);
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/DriverStation.java]: See the comment above about setDriverRumble().

File: src/org/usfirst/frc/team467/robot/Elevator.java [hunk 8]
```diff
@@ -4,6 +4,8 @@
 
 import edu.wpi.first.wpilibj.MotorSafetyHelper;
 import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Elevator.java]: I'm assuming this basically took the master branch version of this file and not looking too closely.

File: src/org/usfirst/frc/team467/robot/Grabber.java [hunk 9]
```diff
@@ -35,31 +54,90 @@ public static Grabber getInstance() {
...
 
 		// Save the previous state and check for current state.
 		hadCube = hasCube;
 		hasCube = os.detectedTarget();
+	}
 
-		if (justGotCube()) {
-			LOGGER.debug("Just got cube, rumbling");
-			DriverStation.getInstance().getNavRumbler().rumble(100, 1.0);
+	public void grab() {
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Grabber.java]: I think it's going to be confusing to have grab() autonomous methods with the same name as the non-autonomous methods with only an argument difference.

File: src/org/usfirst/frc/team467/robot/Grabber.java [hunk 10]
```diff
@@ -35,31 +54,90 @@ public static Grabber getInstance() {
 		return instance;
 	}
 
-	public void grab(double throttle) {
+	public void periodic() {
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Grabber.java]: I'm beginning to think that we should add an AutonomousGrabber class that handles keeping the grabber running for a period of time, etc., and keep the actual Grabber class simpler.

File: src/org/usfirst/frc/team467/robot/MovingAverage.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+
+/**
+ * Utility class for checking current limits over a number of periodic cycles.
+ * This is required to avoid stopping a motor based on the initial current spike at motor start.
+ */
+public class MovingAverage {
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/MovingAverage.java:11]: Is this class being used anywhere? If not, why did we add it?

File: src/org/usfirst/frc/team467/robot/OpticalSensor.java [hunk 12]
```diff
@@ -1,11 +1,14 @@
 package org.usfirst.frc.team467.robot;
+import com.sun.media.jfxmedia.logging.Logger;
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/OpticalSensor.java]: Umm.... what? I don't think you want this.

File: src/org/usfirst/frc/team467/robot/OpticalSensor.java [hunk 13]
```diff
@@ -1,11 +1,14 @@
...
+
 import edu.wpi.first.wpilibj.DigitalInput;
 
 public class OpticalSensor {
 	private static OpticalSensor instance;
 	private DigitalInput di;
 
 	private OpticalSensor() {
+		System.out.println(RobotMap.OPTICAL_CHANNEL);
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/OpticalSensor.java:10]: We should never call println from the robot code.

File: src/org/usfirst/frc/team467/robot/OpticalSensor.java [hunk 14]
```diff
@@ -18,7 +21,12 @@ public static OpticalSensor getInstance() {
 	}
 
 	public boolean detectedTarget() {
-		return di.get();
+//		if (di.get()) {
+//			System.out.println("NO CUBE !!!");
+//		} else {
+//			System.out.println("SEE CUBE !!!!!!");
+//		}
+		return !di.get();
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/OpticalSensor.java]: Please delete the commented out code.

File: src/org/usfirst/frc/team467/robot/Robot.java [hunk 15]
```diff
@@ -35,16 +20,25 @@
...
-
+	private MatchConfiguration matchConfig;
+	private VisionProcessing vision;
 	private Gyrometer gyro;
-
 	private Elevator elevator;
 	private Grabber grabber;
+	private TiltMonitor tiltMonitor;
+
+	int session;
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Unused variable, please remove.

File: src/org/usfirst/frc/team467/robot/Robot.java [hunk 16]
```diff
@@ -35,16 +20,25 @@
...
+
+	/**
+	 * Time in milliseconds
+	 */
+	double time;
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java:42]: This doesn't seem to be used anywhere. Can we remove it?

File: src/org/usfirst/frc/team467/robot/Robot.java [hunk 17]
```diff
@@ -206,18 +135,31 @@ public void teleopPeriodic() {
...
+			turn = 0.0;
 		}
 
-		grabber.grab(driverstation.getGrabThrottle());
-
-		//changed to arcade drive
-		drive.arcadeDrive(left, right, true);
-
-		TiltMonitor.getInstance().periodic();
+		switch (driverstation.getDriveMode()) {
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: I think we don't want this notion of a "drive mode."

File: src/org/usfirst/frc/team467/robot/XBoxJoystick467.java [hunk 18]
```diff
@@ -35,16 +33,16 @@
...
+		a(0),
+		b(1),
+		x(2),
+		y(3),
+		BumperLeft(4),
+		BumperRight(5),
+		back(6),
+		start(7),
+		left(8),
+		right(9);
```
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/XBoxJoystick467.java]: This is wrong, button numbers are 1-based, not 0-based. We discovered this recently.

File: src/org/usfirst/frc/team467/robot/Drive.java (hunk 1 above)
Comment (bryanduerk) [reply src/org/usfirst/frc/team467/robot/Drive.java]: removed

**Comment by bryanduerk** [comment]
All comments addressed, then re-pulled in current master.

**Pull request closed without merging**